internal/file/watch/watch_test.go

```diff
@@ -478,121 +732,123 @@ func Test_watch_Remove(t *testing.T) {
 		dirs []string
 	}
 	type fields struct {
-		w        *fsnotify.Watcher
-		eg       errgroup.Group
-		dirs     map[string]struct{}
-		mu       sync.RWMutex
-		onChange func(ctx context.Context, name string) error
-		onCreate func(ctx context.Context, name string) error
-		onRename func(ctx context.Context, name string) error
-		onDelete func(ctx context.Context, name string) error
-		onWrite  func(ctx context.Context, name string) error
-		onChmod  func(ctx context.Context, name string) error
-		onError  func(ctx context.Context, err error) error
+		w    *fsnotify.Watcher
+		dirs map[string]struct{}
 	}
 	type want struct {
-		err error
+		want *watch
+		err  error
 	}
 	type test struct {
 		name       string
 		args       args
 		fields     fields
 		want       want
-		checkFunc  func(want, error) error
-		beforeFunc func(args)
+		checkFunc  func(want, *watch, error) error
+		beforeFunc func(*testing.T, *fields, args)
 		afterFunc  func(args)
 	}
-	defaultCheckFunc := func(w want, err error) error {
+	defaultCheckFunc := func(w want, got *watch, err error) error {
 		if !errors.Is(err, w.err) {
 			return errors.Errorf("got error = %v, want %v", err, w.err)
 		}
+
+		if got, want := len(got.dirs), len(w.want.dirs); got != want {
+			return errors.Errorf("dirs length = %d, want %d", got, want)
+		}
+
+		for name := range w.want.dirs {
+			if _, ok := got.dirs[name]; !ok {
+				return errors.Errorf("dirs %s is not exists", name)
+			}
+		}
```

Comment:
can we verify the internal `*fsnotify.Watcher` changes?

Reply:
https://github.com/vdaas/vald/pull/526#discussion_r450736130